OpenSCAP/openscap
# Introduce ability to generate kickstarts

This PR introduces ability to generate kickstarts from SCAP content using:

```
oscap xccdf generate fix --fix-type kickstart ds.xml
```

The kickstart will be generated from kickstart snippets in XCCDF rules in the input SCAP content.
The kickstart snippets need to be stored in `<fix>` elements with `system` attribute set to `urn:xccdf:fix:script:kickstart`.

For more details, please read commit messages of each commit.

### Rationale:

We'd like to provide a lightweight system hardening for Anaconda, that is not dependent on UI and does not require user intervention. This should be pretty much similar to Image Builder's experience for users that can not use it for some reason.



### How to test:

1. Build RHEL 9 data stream from the ComplianceAsCode pull request master as of 2024-07-29 or newer.
2. Generate the kickstart `ks.cfg`:
```
./oscap_wrapper xccdf generate fix --profile stig --output ks.cfg --fix-type kickstart ~/work/git/scap-security-guide/build/ssg-rhel9-ds.xml
```
3. Add AppStream repository to the generated kickstart by adding this line (edit the URL)
```
repo --name=extra-repository --baseurl=http://DOWNLOAD_HOSTNAME/released/RHEL-9/9.4.0/AppStream/x86_64/os/
```
4. Serve the kickstart on a HTTP server
```
python3 -m http.server
```
5. Run `virt-manager`, click on Create a new virtual machine. Choose `Network Install`.
6. Provide installation URL for RHEL 9.4: ` http://DOWNLOAD_HOSTNAME/released/RHEL-9/9.4.0/BaseOS/x86_64/os/`
Under the installation URL field, click on `URL options` and insert there `inst.ks=http://192.168.124.1:8000/ks.cfg`. Complete next steps in the wizard.
7. VM will launch. Configure partitioning, continue with installation. Watch Anaconda progress.
8. After installation and reboot, log in to the installed VM and observe state of the system. Read Anaconda logs. Also, you can perform a oscap scan of the installed system.


### Review hints:

Start by reading the proposed user manual text in the PR.
